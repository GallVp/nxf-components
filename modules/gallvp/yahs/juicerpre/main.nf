process YAHS_JUICERPRE {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/yahs:1.2.2--h577a1d6_1':
        'biocontainers/yahs:1.2.2--h577a1d6_1' }"

    input:
    tuple val(meta), path(bam_or_bin)
    path(agp)
    path(fai)

    output:
    tuple val(meta), path("*.txt")          , emit: txt
    tuple val(meta), path("*.assembly")     , emit: assembly        , optional: true
    tuple val(meta), path("*.assembly.agp") , emit: assembly_agp    , optional: true
    tuple val(meta), path("*.liftover.agp") , emit: liftover_agp    , optional: true
    tuple val(meta), path("*.sizes")        , emit: sizes           , optional: true
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def sizes_cmd = '-a' in "$args" ? "sed -n 's|PRE_C_SIZE: \\(.*\\)|\\1|p' ${prefix}.stdout > ${prefix}.sizes" : ''
    """
    juicer pre \\
        $args \\
        $bam_or_bin \\
        $agp \\
        $fai \\
        -o ${prefix} \\
        2>| >(tee ${prefix}.stdout >&2)

    $sizes_cmd

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        yahs: \$(yahs --version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        yahs: \$(yahs --version)
    END_VERSIONS
    """
}
