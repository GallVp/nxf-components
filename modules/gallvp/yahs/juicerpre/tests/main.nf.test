nextflow_process {

    name "Test Process YAHS_JUICERPRE"
    script "../main.nf"
    config "./nextflow.config"
    process "YAHS_JUICERPRE"

    tag "modules"
    tag "modules_gallvp"
    tag "yahs"
    tag "yahs/juicerpre"

    test("hongyang - hic - bam") {

        when {

            params {
                juicerpre = ''
            }

            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - bam - assembly") {

        when {

            params {
                juicerpre = '-a'
            }

            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - bam - stub") {

        options '-stub'

        when {

            params {
                juicerpre = ''
            }

            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hongyang - hic - bam - assembly - stub") {

        options '-stub'

        when {

            params {
                juicerpre = '-a'
            }

            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.bam' , checkIfExists: true)
                ]

                input[1] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp' , checkIfExists: true)
                input[2] = file( 'https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.fasta.fai' , checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}