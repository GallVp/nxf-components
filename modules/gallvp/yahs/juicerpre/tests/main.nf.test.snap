{
    "hongyang - hic - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,461ec5ed8e977fd0276415ca0e490ce2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,2154e1e2f2c12dd43765bc35a2003826"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly.agp:md5,1389ded06e1056dd03d44a849788c4ad"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.liftover.agp:md5,a98e3ca7dc4aea3523197faba02c3d72"
                    ]
                ],
                "4": [
                    "versions.yml:md5,81b6636241b0f8f09004f6b92dcab8a7"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,2154e1e2f2c12dd43765bc35a2003826"
                    ]
                ],
                "assembly_agp": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly.agp:md5,1389ded06e1056dd03d44a849788c4ad"
                    ]
                ],
                "liftover_agp": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.liftover.agp:md5,a98e3ca7dc4aea3523197faba02c3d72"
                    ]
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,461ec5ed8e977fd0276415ca0e490ce2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,81b6636241b0f8f09004f6b92dcab8a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-22T11:06:26.842579"
    },
    "hongyang - hic - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    "versions.yml:md5,81b6636241b0f8f09004f6b92dcab8a7"
                ],
                "assembly": [
                    
                ],
                "assembly_agp": [
                    
                ],
                "liftover_agp": [
                    
                ],
                "txt": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,81b6636241b0f8f09004f6b92dcab8a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-22T13:40:28.868116"
    }
}